src/Microsoft.ML.Core/Data/ColumnTypeExtensions.cs

```diff
@@ -118,6 +120,8 @@ public static PrimitiveDataViewType PrimitiveTypeFromKind(DataKind kind)
                 return DateTimeDataViewType.Instance;
             if (kind == DataKind.DZ)
                 return DateTimeOffsetDataViewType.Instance;
+            if (kind == DataKind.UG)
```

Comment:
>DataKind.UG [](start = 24, length = 11)

Since `DataKind.UG` is only for these `RowId` instances, perhaps we should rename `DataKind.UG` to `DataKind.RowId`. When I first ran across `UG`, I thought it was for holding large unsigned integers.... Could be a more user friendly nomenclature, I guess.

Reply:
@wschin is currently working on #2661, which makes `DataKind.UG` internal. So it wouldn't be user facing.